### PR TITLE
jce for java7 and 8

### PR DESCRIPTION
**Pull request by neilmillard-msm**

### CODE REVIEW

**Comment by neilmillard-msm** [comment]
java currently looks at http://s3-eu-west-1.amazonaws.com/msm-public-repo/java for its source. we could simplify by putting jce source in the same location as the java package

File: manifests/package.pp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class java::package{
+
+  package{ 'unzip':
+    ensure => installed,
+  }
+
+  package{ 'zip':
+    ensure => installed,
```
Comment (chrisjomaron) [inline manifests/package.pp]: do we require zip, or just unzip?

File: manifests/init.pp [hunk 2]
```diff
@@ -1,7 +1,8 @@
 #
-#
+# need to override jce_url as direct download not permitted
 class java (
   $source_url          = 'https://download.oracle.com/otn-pub/java/jdk/7u51-b13',
+  $jce_repo_url        = 'http://download.oracle.com/otn-pub/java/jce',
```
Comment (erd-msm) [inline manifests/init.pp]: any point to this default that doesn't work ?

Comment (neilmillard-msm) [inline manifests/init.pp]: gives the user a clue for where to get the source from

Comment (erd-msm) [inline manifests/init.pp]: not really, the link is useless. A comment would be better:
# original download from http://download.oracle.com/otn-pub/java/jce
# link is no good without authentication token,
# goto http://download.oracle.com and search for jce

File: manifests/init.pp [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #
-#
+# need to override jce_url as direct download not permitted
+# original download from http://download.oracle.com/otn-pub/java/jce
+# link is no good without authentication token,
+# goto http://download.oracle.com and search for jce
 class java (
+  $jce_repo_url,
```
Comment (erd-msm) [inline manifests/init.pp]: you may as well put in the artifactory URL, The wget path in the jce.pp is specific to MSM artifactory. In for a penny etc...

**Pull request merged**